### DISCUSSION TITLE
BOINC Client will not see Intel GPU after update to 6.8.1

### DISCUSSION THREAD

**Discussion by debianmain1** [BOINC/boinc#5722] [Q&A]
**Describe the bug**
Boinc client no longer sees the Intel GPU after the update to Kernel 6.8.1   Downgrading to 6.7.9 allows the Intel GPU to be seen.

**Steps To Reproduce**
1. Upgrade from Kernel 6.7.9 to 6.8.1 with an Intel GPU installed.

**Expected behavior**
Boinc finding Intel GPU as per normal.

**Additional context**
I believe that this problem is with the addition of the Xe driver in the 6.8 kernel line. I have tried to blacklist the Xe driver module from loading without success.  The Xe driver "should" be made available to BOINC due to better handling of hardware. The i915 is very old & will not have the performance of the Xe driver once it matures. 

Ideally, BOINC being able to handle both drivers would be optimal. In the short term, defaulting to the i915 driver & ignoring the Xe driver would allow BOINC to continue to use Intel GPUs.

Method of blacklist:  In /etc/modprobe.d I added:  blacklist xe  &  options xe modeset=0  to the blacklist.conf file. no change was noted - BOINC still would not see the Intel GPU.

**System Information** In screenshot below

 - BOINC Version:  7.24.1

Kernel 6.7.9-zen1-1-zen:

03:00.0 VGA compatible controller: Intel Corporation DG2 [Arc A750] (rev 08) (prog-if 00 [VGA controller])
	Subsystem: Intel Corporation DG2 [Arc A750]
	Flags: bus master, fast devsel, latency 0, IRQ 250
	Memory at 82000000 (64-bit, non-prefetchable) [size=16M]
	Memory at 4000000000 (64-bit, prefetchable) [size=8G]
	Expansion ROM at 83000000 [disabled] [size=2M]
	Capabilities: <access denied>
	Kernel driver in use: i915
	Kernel modules: i915

Kernel 6.8.1 -zen1-1-zen:

03:00.0 VGA compatible controller: Intel Corporation DG2 [Arc A750] (rev 08) (prog-if 00 [VGA controller])
	Subsystem: Intel Corporation DG2 [Arc A750]
	Flags: bus master, fast devsel, latency 0, IRQ 250
	Memory at 82000000 (64-bit, non-prefetchable) [size=16M]
	Memory at 4000000000 (64-bit, prefetchable) [size=8G]
	Expansion ROM at 83000000 [disabled] [size=2M]
	Capabilities: <access denied>
	Kernel driver in use: i915
	Kernel modules: i915, xe

Note the addition of the "xe" Intel driver in kernel modules.

**Screenshots**
![Screenshot from 2024-03-17 08-02-42](https://github.com/BOINC/boinc/assets/72761/7f898ba4-b2ca-4321-94e2-772835f9679d)

**Comment by debianmain1** [comment]
Any thoughts on how I can use BOINC with >6.8 until a fix is pushed? I'm fine with a manual intervention. I'm doing testing on the Xe driver in EndeavourOS/Arch & would like to avoid booting between the LTS & main kernels and ideally try to use the Xe driver for work units. A couple of projects are using the Windows Xe driver & it would be VERY interesting to see if the Linux Xe driver is compatible.

**Comment by debianmain1** [comment]
OK---Intel-Compute has been patched in Arch Linux. This now allows BOINC 8.1 to see the Intel GPU. As far as I know, this refers only to the i915 driver. Versions 7.24 & 8.0 also work with this patch. I still think there needs to be a "switch" between i915 & Xe driver usage.

Arch Commit:  https://gitlab.archlinux.org/archlinux/packaging/packages/intel-compute-runtime/-/commit/f38c305e75dbb341eeeff5faeb431d8952af1b7d#1e01a16726e6e3e5b0808c37cfba79d1ee7d8801

**Comment by debianmain1** [comment]
I will be testing the Xe driver in Arch as soon as Mesa is updated to 24.1 (about 1 month). I will report here as to the results.

**Comment by Toby-Broom** [comment]
i got it to detect with an A380 after installing the compute runtime e.g. 

https://github.com/intel/compute-runtime/releases/tag/24.13.29138.7

I think you could do with these now https://dgpu-docs.intel.com/driver/client/overview.html

**Comment by debianmain1** [comment]
Yes...
I noted that 2 posts above your post.....I am almost ready to test the Xe driver (next Mesa update) & will report then if BOINC works with it.

**Comment by debianmain1** [comment]
OK...I just was able to run the Xe driver for the first time in Arch Linux...Intel-compute-runtime is current & Mesa is current. BOINC Manager will not see the Xe driver...when I reverted to the i915 driver, BOINC manager found the Intel driver without a problem.

**Comment by debianmain1** [comment]
Thank you for combining #5639 ...I really feel that this needs to be available as the Xe driver is coming along very quickly....see:  https://www.phoronix.com/news/Linux-6.10-Intel-Xe-DRM-Patches

**Comment by AenBleidd** [comment]
Few comments from my side: without very deep analysis:
1. BOINC uses opencl (in this particular case) to identify the GPU. If opencl fails doing that - BOINC fails to detect GPU as well. And this is by design, and won't be changed
2. Looks like this issue will be resolved by itself, because fixed version of opencl (or whatever) is currently landing to the experimental builds of Linux distros.

For sure, we will need to monitor this situation, but I don't think we should do anything right now with this.

**Comment by debianmain1** [comment]
Thank you so much for the clarification....I will start tracking OpenCl updates also and keep this updated.

**Comment by AenBleidd** [comment]
Converting this to discussion, since there is nothing that should be done on the BOINC side
